### PR TITLE
Better inval

### PR DESCRIPTION
**Pull request by nmsimons**

### CODE REVIEW

**Review by ChumpChief — DISMISSED** [review]
Love the improvements in item counter - brainstorm is still doing an invalidations-like approach in a few places which would be good to try to get away from if you can.  Overall these are improvements though so I'm going ahead with approval, lmk if you'd like me to take another look later.

**Review by ChumpChief — APPROVED** [review]
Nice improvements :)

File: brainstorm/src/react/groupux.tsx [hunk 1]
```diff
@@ -73,9 +82,9 @@ export function GroupView(props: {
 				return;
 			}
 
-			const droppedItem = item;
-			if (Tree.is(droppedItem, Group) || Tree.is(droppedItem, Note)) {
-				moveItem(droppedItem, props.parent.indexOf(props.group), props.parent);
+			if (Tree.is(item, Group) || Tree.is(item, Note)) {
+				console.log("drop");
```
[the remark this answers is not in the document]
Comment (nmsimons) [reply brainstorm/src/react/groupux.tsx]: fixed

File: item-counter-spe/webpack.config.cjs [hunk 2]
```diff
@@ -52,7 +52,7 @@ module.exports = {
 		// This line is VERY important for VS Code debugging to attach properly
 		// Tamper with it at your own risks
 		devtoolModuleFilenameTemplate: "[absolute-resource-path]",
-		clean: true,
+		clean: false,
```
[the remark this answers is not in the document]
Comment (nmsimons) [reply item-counter-spe/webpack.config.cjs:55]: Fixes hot reload. I'm sure there is a better way but I haven't found it.

File: brainstorm/src/schema/app_schema.ts [hunk 3]
```diff
@@ -41,12 +41,12 @@ export class Note extends sf.object(
 	},
 ) {
 	// Update the note text and also update the timestamp in the note
-	public updateText(text: string) {
+	public updateText = (text: string) => {
```
[the remark this answers is not in the document]
Comment (nmsimons) [reply brainstorm/src/schema/app_schema.ts]: fixed

File: brainstorm/src/react/canvasux.tsx [hunk 4]
```diff
@@ -41,13 +41,13 @@ export function Canvas(props: {
 	setSaved: (arg: boolean) => void;
 	setFluidMembers: (arg: string[]) => void;
 }): JSX.Element {
-	const [invalidations, setInvalidations] = useState(0);
+	const [, setItemArray] = useState(props.items.root.map((item) => item));
```
[the remark this answers is not in the document]
Comment (nmsimons) [reply brainstorm/src/react/canvasux.tsx]: fixed

File: brainstorm/src/react/groupux.tsx [hunk 5]
```diff
@@ -106,39 +115,49 @@ export function GroupView(props: {
...
+function GroupToolbar(props: {
+	name: string;
+	changeName: (name: string) => void;
+	deletePile: () => void;
+}): JSX.Element {
 	return (
 		<div className="flex flex-row justify-between">
-			<GroupName pile={props.pile} />
-			<DeletePileButton pile={props.pile} items={props.parent} />
+			<GroupName {...props} />
```
[the remark this answers is not in the document]
Comment (nmsimons) [reply brainstorm/src/react/groupux.tsx:154]: next time

**Review by ChumpChief — APPROVED** [review]

**Pull request merged**